lib/publiccloud/instance.pm

```diff
@@ -481,6 +481,12 @@ sub softreboot {
     if ($tunneled) {
         select_console('tunnel-console', await_console => 0);
         ssh_interactive_leave();
+        my $i = 5;
+        while (script_run(sprintf('ssh -O check %s@%s', $args{username}, $self->public_ip)) == 0 && $i != 0) {
+            script_run(sprintf('ssh -O exit %s@%s', $args{username}, $self->public_ip));
+            $i = $i - 1;
+            sleep 5;
+        }
```

Comment:
It seems more natural to me to just do a `for` loop here, but you may skip this suggestion.

```suggestion
        for (1 .. 5) {
            last if (script_run(sprintf('ssh -O check %s@%s', $args{username}, $self->public_ip)) != 0);
            script_run(sprintf('ssh -O exit %s@%s', $args{username}, $self->public_ip));
            sleep 5;
        }
```

Reply:
agree with Ricardo "for loop" looks better here